app/models/server.rb

```diff
@@ -71,301 +66,79 @@ class Server < ActiveRecord::Base
+  non_versioned_columns << 'port'
+  non_versioned_columns << 'rcon'
+  non_versioned_columns << 'password'
+  non_versioned_columns << 'irc'
+  non_versioned_columns << 'user_id'
+  non_versioned_columns << 'official'
+  non_versioned_columns << 'domain'
+  non_versioned_columns << 'reservation'
+  non_versioned_columns << 'recording'
+  non_versioned_columns << 'idle'
+  non_versioned_columns << 'default_id'
+  non_versioned_columns << 'active'
+  non_versioned_columns << 'recordable_type'
+  non_versioned_columns << 'recordable_id'
+
+  def domains
+    {DOMAIN_HLTV => "HLTV", DOMAIN_HLDS => "NS Server", DOMAIN_NS2 => "NS2 Server"}
+  end
 
-def can_update? cuser
-  cuser and cuser.admin? or user == cuser
-end
+  def to_s
+    name
+  end
 
-def can_destroy? cuser
-  cuser and cuser.admin?
-end
+  def addr
+    ip + ":" + port.to_s
+  end
 
-def self.refresh
-  servers = ""
-  Server.hlds.active.all.each do |server|
-    servers << " -a2s " + server.ip + ":" + server.port.to_s
+  def set_category
+    self.category_id = (domain == DOMAIN_NS2 ? 45 : 44 )
   end
 
-  file = File.join(Rails.root, TMPFILE)
-  system "#{QSTAT} -xml #{servers} | grep -v '<name>' > #{file}"
+  def is_free time
+    challenges.around(time).pending.count == 0 and matches.around(time).count == 0
+  end
 
-  doc = REXML::Document.new(File.new(file).read)
-  doc.elements.each('qstat/server') do |server|
-    hostname = server.elements['hostname'].text.split(':', 2)
-    if s = Server.active.first(:conditions => {:ip => hostname[0], :port => hostname[1]})
-      if server.elements.include? 'map'
-        s.map = server.elements['map'].text
-        s.players = server.elements['numplayers'].text.to_i
-        s.max_players = server.elements['maxplayers'].text.to_i
-        s.ping = server.elements['ping'].text
-        s.map = server.elements['map'].text
-        s.save
-        s.make_stats
-      end
-    end
+  def can_create? cuser
+    cuser
   end
 
-  servers = ""
-  Server.hltvs.reserved.each do |server|
-    servers << " -a2s #{server.reservation}"
+  def can_update? cuser
+    cuser and cuser.admin? or user == cuser
   end
 
-  doc = REXML::Document.new(`#{QSTAT} -xml #{servers} | grep -v '<name>'`)
-  doc.elements.each('qstat/server') do |server|
-    hostname = server.elements['hostname'].text.split(':', 2)
-    if s = Server.hltvs.reserved.first(:conditions => {:ip => hostname[0], :port => hostname[1]})
-      if server.elements['numplayers'].text.to_i > 0
-        s.update_attribute :idle, DateTime.now
-      elsif (s.idle + HLTV_IDLE).past?
-        s.reservation = nil
-        s.save
-      end
-    end
+  def can_destroy? cuser
+    cuser and cuser.admin?
   end
-end
 
-def self.move addr, newaddr, newpwd
-  self.hltvs.all(:conditions => {:reservation => addr}).each do |hltv|
-    hltv.reservation = newaddr
-    hltv.pwd = newpwd
-    hltv.save!
+  def self.move addr, newaddr, newpwd
+    self.hltvs.all(:conditions => {:reservation => addr}).each do |hltv|
```

Comment:
Use the new Ruby 1.9 hash syntax.<br>Redundant `self` detected.<br>Space inside { missing.<br>Space inside } missing.